### PR TITLE
feat: announce: mattermost

### PR DESCRIPTION
**Pull request by cpanato**
Add mattermost to the announcements 

Test Announcement:
Screenshot 2021-09-28 at 14 58 12

### CODE REVIEW

File: internal/pipe/mattermost/mattermost.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/apex/log"
+	"github.com/caarlos0/env/v6"
+	"github.com/mattermost/mattermost-server/v5/model"
```
Comment (caarlos0) [inline internal/pipe/mattermost/mattermost.go:12]: seems like we're adding a lot of deps just to use the model package... maybe its better to copy that struct into our code instead?

File: internal/pipe/mattermost/mattermost.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+		return fmt.Errorf("announce: failed to announce to mattermost: %w", err)
+	}
+
+	return nil
+}
+
+func PostWebhook(ctx *context.Context, url string, msg *model.IncomingWebhookRequest) error {
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, bytes.NewReader([]byte(msg.ToJson())))
+	if err != nil {
+		return errors.Wrap(err, "failed new request")
```
Comment (caarlos0) [inline internal/pipe/mattermost/mattermost.go:95]:
```suggestion
		return fmt.Errorf("failed new request: %w", err)
```

just to be consistent with other pkgs :)

File: internal/pipe/mattermost/mattermost.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+	err = PostWebhook(ctx, cfg.Webhook, wm)
+	if err != nil {
+		return fmt.Errorf("announce: failed to announce to mattermost: %w", err)
+	}
+
+	return nil
+}
+
+func PostWebhook(ctx *context.Context, url string, msg *model.IncomingWebhookRequest) error {
```
Comment (caarlos0) [inline internal/pipe/mattermost/mattermost.go:92]: don't need to be exposed:

```suggestion
func postWebhook(ctx *context.Context, url string, msg *model.IncomingWebhookRequest) error {
```

**Review by caarlos0 — CHANGES_REQUESTED** [review]
Hey, thanks for the PR!

added a few comments!

File: internal/pipe/mattermost/mattermost.go (hunk 1 above)
Comment (cpanato) [reply internal/pipe/mattermost/mattermost.go:12]: sounds great to me!

Comment (cpanato) [reply internal/pipe/mattermost/mattermost.go:12]: will make the changes, thanks for your feedback and review!

File: internal/pipe/mattermost/mattermost_test.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+		require.Equal(t, "Honk v1.0.0 is out! Check it out at https://github.com/honk/honk/releases/tag/v1.0.0", rc.Attachments[0].Text)
+
+		w.WriteHeader(200)
+		_, err = w.Write([]byte{})
+		require.NoError(t, err)
+	}))
+	defer ts.Close()
+
+	ctx := context.New(config.Project{
+		ProjectName: "Honk",
```
Comment (dirien) [inline internal/pipe/mattermost/mattermost_test.go:83]: haha nice one! ❤️

**Review by dirien — APPROVED** [review]
Da ballert der gute @cpanato mal die Mattermost integration raus! So cool! ❤️

File: internal/pipe/mattermost/mattermost.go (hunk 3 above)
Comment (cpanato) [reply internal/pipe/mattermost/mattermost.go]: thanks again @caarlos0 did the changes you requested

**Comment by caarlos0** [comment]
Merged, thanks again @cpanato ❤️

**Pull request merged**